### DISCUSSION TITLE
Page exists in both Microsoft.WinUI

### DISCUSSION THREAD

**Discussion by limeniye** [unoplatform/uno#13678] [Q&A] [answered]
**Uno.WinUI 5.0.0-dev.2593**
**.net 8**

![image](https://github.com/unoplatform/uno/assets/69314237/c78a1cf4-9028-4bb6-9926-eb0bdb30d2ae)

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	UXAML0001	Unable to find type Microsoft.UI.Xaml.ResourceDictionary	UnoAppGridSplitter	R:\ForDelete\UnoAppGridSplitter\UnoAppGridSplitter\UnoAppGridSplitter\CSC	1	

Severity	Code	Description	Project	File	Line	Suppression State
Error	UXAML0001	Processing failed for file R:\ForDelete\UnoAppGridSplitter\UnoAppGridSplitter\UnoAppGridSplitter\AppResources.xaml (System.InvalidOperationException: Unable to find type Microsoft.UI.Xaml.ResourceDictionary	UnoAppGridSplitter	R:\ForDelete\UnoAppGridSplitter\UnoAppGridSplitter\UnoAppGridSplitter\CSC	1	

Severity	Code	Description	Project	File	Line	Suppression State
Error	UXAML0001	Processing failed for file R:\ForDelete\UnoAppGridSplitter\UnoAppGridSplitter\UnoAppGridSplitter\MainPage.xaml (System.InvalidOperationException: Unable to find type Microsoft.UI.Xaml.Application
   at Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGeneration.<>c__DisplayClass172_0.<GetMandatorySymbolAsLazy>b__0()
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at Uno.UI.SourceGenerators.XamlGenerator.XamlFileGenerator.IsApplication(INamedTypeSymbol symbol)
   at Uno.UI.SourceGenerators.XamlGenerator.XamlFileGenerator.BuildInitializeComponent(IndentedStringBuilder writer, XamlObjectDefinition topLevelControl, INamedTypeSymbol controlBaseType)
   at Uno.UI.SourceGenerators.XamlGenerator.XamlFileGenerator.InnerGenerateFile()
   at Uno.UI.SourceGenerators.XamlGenerator.XamlFileGenerator.GenerateFile())	UnoAppGridSplitter (net8.0-windows10.0.19041)	R:\ForDelete\UnoAppGridSplitter\UnoAppGridSplitter\UnoAppGridSplitter\UnoAppGridSplitter.csproj	1	Active
```

[UnoAppPageExistsInBoth.zip](https://github.com/unoplatform/uno/files/12649916/UnoAppPageExistsInBoth.zip)

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report. It seems that nuget has changed props ordering and it's breaking our winui detection. https://github.com/unoplatform/uno/pull/13700 will improve this situation.

**Reply by limeniye** [reply]
Thank you for information!
I'am waiting for updates.

**Comment by tibitoth** [comment]
What is the current fix for this? I've just updated all my nuget packages in my .NET 7 app (WinAppSDK and Uno as well) and got this error.

**Reply by jeromelaban** [reply]
Thanks for the update. Could you tell which package versions you are using?

**Reply by tibitoth** [reply]
@jeromelaban I used only these packages:

```xml
    <PackageReference Include="Uno.Extensions.Hosting.WinUI" Version="2.5.11" />
    <PackageReference Include="Uno.Extensions.Logging.WinUI" Version="2.5.11" />
    <PackageReference Include="Uno.Extensions.Logging.Serilog" Version="2.5.11" />
    <PackageReference Include="Uno.Extensions.Navigation.WinUI" Version="2.5.11" />
```

Based on your [suggestion](https://github.com/unoplatform/uno/pull/13700#issuecomment-1744765293) I've just added `<PackageReference Include="Uno.WinUI" Version="4.10.26" />` and it works now. Maybe some transitive reference's version could be too low.

**Reply by jeromelaban** [reply]
Thanks for the update. Indeed, this is likely the cause. Adding an explicit reference to the latest uno is best to fix those issue.

**Reply by jeromelaban** [reply]
In order to dig a bit further into this, how did you create your project? Did you use dotnet new or the VS wizard?

**Reply by tibitoth** [reply]
started from a regular winui3 blank app and tried to use only the extension packages because currently I only need hosting, logging and navigation from uno

**Reply by tibitoth** [reply]
I would not think that I need to reference `Uno.Winui` and this works with winappsdk 1.3, only broke with 1.4

**Reply by jeromelaban** [reply]
I see. Thanks, it makes sense. We may need to adjust references for that particular scenario. /cc @nickrandolph
